src/dotnet/APIView/APIViewWeb/Languages/CLanguageService.cs

```diff
@@ -0,0 +1,469 @@
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.Json;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using ApiView;
+using APIView;
+
+namespace APIViewWeb
+{
+    public class CLanguageService : ILanguageService
+    {
+        private const string CurrentVersion = "2";
+        private static Regex _typeTokenizer = new Regex("\\w+|[^\\w]+", RegexOptions.Compiled | RegexOptions.IgnoreCase);
+        private static HashSet<string> _keywords = new HashSet<string>()
+        {
+            "auto",
+            "break",
+            "bool",
+            "case",
+            "char",
+            "const",
+            "continue",
+            "default",
+            "do",
+            "double",
+            "else",
+            "enum",
+            "extern",
+            "float",
+            "for",
+            "goto",
+            "if",
+            "inline",
+            "int",
+            "long",
+            "register",
+            "restrict",
+            "return",
+            "short",
+            "signed",
+            "sizeof",
+            "static",
+            "struct",
+            "switch",
+            "typedef",
+            "union",
+            "unsigned",
+            "void",
+            "volatile",
+            "while",
+            "_Alignas",
+            "_Alignof",
+            "_Atomic",
+            "_Bool",
+            "_Complex",
+            "_Generic",
+            "_Imaginary",
+            "_Noreturn",
+            "_Static_assert",
+            "_Thread_local"
+        };
+
+        public string Name { get; } = "C";
+
+        public bool IsSupportedExtension(string extension) => string.Equals(extension, ".ast", comparisonType: StringComparison.OrdinalIgnoreCase);
+
+        public bool CanUpdate(string versionString) => true;
+
+        public async Task<CodeFile> GetCodeFileAsync(string originalName, Stream stream, bool runAnalysis)
+        {
+            MemoryStream astStream = new MemoryStream();
+            await stream.CopyToAsync(astStream);
+
+            return BuildNodes(originalName, astStream);
+        }
+
+        private static CodeFile BuildNodes(string originalName, MemoryStream astStream)
+        {
+            Span<byte> ast = astStream.ToArray();
+
+            CodeFileTokensBuilder builder = new CodeFileTokensBuilder();
+            Dictionary<string, NavigationItem> navigation = new Dictionary<string, NavigationItem>();
+
+            while (ast.Length > 2)
+            {
+                Utf8JsonReader reader = new Utf8JsonReader(ast);
+
+                var astNode = JsonSerializer.Deserialize<CAstNode>(ref reader);
+                ast = ast.Slice((int)reader.BytesConsumed);
+
+                var queue = new Queue<CAstNode>(astNode.inner);
+                var types = new HashSet<string>();
+
+                foreach (var node in queue)
+                {
+
+                    if (node.kind == "TypedefDecl")
```

Comment:
Should this be case insensitive check?

Reply:
They seem to be consistent in output files.